document-generator-prosecution/src/main/java/uk/gov/companieshouse/document/generator/prosecution/ProsecutionDocumentInfoService.java

```diff
@@ -0,0 +1,180 @@
+package uk.gov.companieshouse.document.generator.prosecution;
+
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.document.generator.interfaces.DocumentInfoService;
+import uk.gov.companieshouse.document.generator.interfaces.exception.DocumentInfoException;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoRequest;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfoResponse;
+import uk.gov.companieshouse.document.generator.prosecution.UltimatumDocumentInfoBuilderProvider.UltimatumDocumentInfoBuilder;
+import uk.gov.companieshouse.document.generator.prosecution.tmpclient.ProsecutionCase;
+import uk.gov.companieshouse.document.generator.prosecution.tmpclient.ProsecutionClient;
+import uk.gov.companieshouse.document.generator.prosecution.tmpclient.SdkException;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+/**
+ * The Prosecution module of Document Generator. When a request with a matching URI is received by
+ * Document Generator, it is forwarded to this module (see matching, below). The URI is further
+ * processed by DocumentGenerator to see which kind of document is to be generated. Different
+ * documents require different template variables and so have different methods to fill in those
+ * template values and other metadata in a {@link DocumentInfoResponse} that is returned to Document
+ * Generator. Document Generator will then use this metadata to determine where to send the template
+ * values and other metadata.
+ * </p>
+ * 
+ *
+ * <h3>Matching</h3> The enum uk.gov.companieshouse.document.generator.api.document.DocumentType,
+ * within the document generator api project, is used to determine whether or not to steer a request
+ * to this service, based on whether the URI matches the regex in
+ * uk.gov.companieshouse.document.generator.api.document.DocumentType.PROSECUTION. </br>
+ * Matching URIs must take the form:
+ * <ul>
+ * <li>Ultimatum: <code>/prosecution/ultimatum{urlOfProsecutionCase}</code></br>
+ * e.g.
+ * <code>/prosecution/ultimatum/company/00066516/prosecution-cases/dd908afef2b8529aab3c680239f5d06717113634</code>
```

Comment:
We shouldn't be using real data as examples in the code. "00066516" looks like a real company number?

Reply:
fixed